sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -591,4 +597,68 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       })
       .getOne();
   }
+
+  /**
+   * Get all offerings that were were requested for change
+   * i.e in Awaiting Approval status.
+   * @returns all offerings that were requested for change.
+   */
+  async getOfferingChangeRequests(): Promise<EducationProgramOffering[]> {
+    return this.repo
+      .createQueryBuilder("offerings")
+      .select([
+        "offerings.id",
+        "offerings.name",
+        "offerings.submittedDate",
+        "offerings.precedingOffering",
+        "educationProgram.id",
+        "institutionLocation.name",
+        "institution.legalOperatingName",
+        "institution.operatingName",
+      ])
+      .innerJoin("offerings.educationProgram", "educationProgram")
+      .innerJoin("offerings.institutionLocation", "institutionLocation")
+      .innerJoin("institutionLocation.institution", "institution")
+      .where("offerings.offeringStatus = :offeringStatus", {
+        offeringStatus: OfferingStatus.AwaitingApproval,
+      })
+      .getMany();
+  }
+
+  /**
+   * For a given offering which is requested as change
+   * get the summary of it's actual(preceding) offering.
+   * @param offeringId offering id of actual offering.
+   * @returns preceding offering summary.
+   */
+  async getPrecedingOfferingSummary(
+    offeringId: number,
+  ): Promise<PrecedingOfferingSummaryModel> {
+    const query = this.dataSource
+      .getRepository(Application)
+      .createQueryBuilder("application")
+      .select("count(application.id)", "applicationsCount")
+      .addSelect("offering.id", "offeringId")
+      .innerJoin("application.currentAssessment", "assessment")
+      .innerJoin("assessment.offering", "offering")
+      .where("application.applicationStatus != :cancelledStatus", {
+        cancelledStatus: ApplicationStatus.cancelled,
+      })
+      .andWhere((qb) => {
```

Comment:
It is not clear to me why we need this sub-query. Is it not possible to move the below inner join to the above query?
````typescript
.innerJoin("offering.precedingOffering", "precedingOffering")
````